docs/development/quickstart.rst

```diff
@@ -0,0 +1,335 @@
+
+There are many good options for doing this, including a number of virtual environment
+managers (e.g., the Python standard library `venv <https://docs.python.org/3/library/venv.html>`_
+module). Users who have a preference for a particular virtual environment manager are
+encouraged to use it!
+
+For this quickstart guide we use the `conda <https://docs.conda.io/en/latest/>`_ package
+manager provided by `miniforge <https://github.com/conda-forge/miniforge>`_. This is a
+popular choice and generally works well, especially for newcomers. It is easy to install
+and use on all platforms and it makes it easy to install different Python versions which
+can be useful for testing.
+
+Install miniforge and conda
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you do not already have ``conda`` installed, `download and install miniforge
+<https://github.com/conda-forge/miniforge/blob/main/README.md>`_. The details depend on
+your system but the end result is to provide a ``conda`` executable that you can use
+to create and manage isolated Python environments.
+
+Now create and activate an ``astropy-dev`` conda environment using the following::
+
+   conda create -n astropy-dev python graphviz
+   conda activate astropy-dev
+
+Note the ``graphviz`` package is required for building the documentation.
+
+Install the development version of astropy
+------------------------------------------
+
+Now you can install the development version of astropy into your new environment. This
+will install the latest version of astropy from your local git repo, along with
+all the dependencies needed to build and fully test astropy::
+
+   python -m pip install --editable '.[dev_all]'
+
+**Checking the build**
+
+At this point you should be able to import astropy from your locally built version::
+
+   python -c 'import astropy; astropy.system_info()'
+
+Next you may want to try running some or all of the ``astropy`` unit tests.
+Running the full test suite can take a few minutes, so you may want to start with a
+single sub-package (e.g. :ref:`astropy-coordinates`)::
+
+
+   # run a sub set of the test suite
+   pytest astropy/coordinates
+
+   # or the whole suite
+   pytest
+
+Details on running and writing tests can be found in the :ref:`testing-guidelines`
+section.
+
+.. _contributing.pre-commit:
+
+Install pre-commit
+------------------
+
+This is optional, but *highly recommended*. `Pre-commit <https://pre-commit.com/>`_ is a
+tool that runs a number of :ref:`Continuous Integration (CI) <contributing.ci>` checks
+(e.g. code formatting) on your code before you commit it. If you skip this step then it
+is likely that one or more of those CI checks will fail when you make a pull request,
+resulting in lost time (yours and CI resources).
+
+Installation is straightforward. From the root of the astropy repository, run::
+
+    pre-commit install
+
+Now all of the styling checks will be
+run each time you commit changes without your needing to run each one manually.
```

Comment:
```suggestion
run each time you commit changes without your needing to run each one manually.
These same checks will also be run on GitHub once you have made or updated your pull request
(see below) and need to pass before your changes can be accepted.
```
I still managed on first pass to overread the pre-commit section here, so can only help to give it more visibility, I hope.

Reply:
I propose you replace the suggestion of "(see below)" with explicit intersphinx to be less vague and more future-proof of any further re-org.

Reply:
This will depend a bit on the pre-commit.ci reference being added there.

Reply:
Addressed in [1e42733](https://github.com/astropy/astropy/pull/16561/commits/1e427332baca08db47bf10c452802732f6a69682) with slight changes.